From 1613f18ca14f76e3f2bc8c19c2cc54a946e8dfed Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:18 +0900
Subject: [PATCH 0937/1083] Revert "drm/rcar-du: Use devm_ioremap_resource()"

This reverts commit a2109c3623cea2747bd7164dd1b943c074e6b5eb.
(cherry picked from commit daefe5f2eea9db96ce30cb6f7856f82f450016b1)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 26323bc..e509c3f 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -113,6 +113,7 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 	struct platform_device *pdev = dev->platformdev;
 	struct rcar_du_platform_data *pdata = pdev->dev.platform_data;
 	struct rcar_du_device *rcdu;
+	struct resource *ioarea;
 	struct resource *mem;
 	int ret;
 
@@ -139,7 +140,15 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 		return -EINVAL;
 	}
 
-	rcdu->mmio = devm_ioremap_resource(&pdev->dev, mem);
+	ioarea = devm_request_mem_region(&pdev->dev, mem->start,
+					 resource_size(mem), pdev->name);
+	if (ioarea == NULL) {
+		dev_err(&pdev->dev, "failed to request memory region\n");
+		return -EBUSY;
+	}
+
+	rcdu->mmio = devm_ioremap_nocache(&pdev->dev, ioarea->start,
+					  resource_size(ioarea));
 	if (rcdu->mmio == NULL) {
 		dev_err(&pdev->dev, "failed to remap memory resource\n");
 		return -ENOMEM;
-- 
1.8.3.2

